File: C++/781. Rabbits in Forest.cpp
class Solution {
public:
    int numRabbits(vector<int>& answers) {
        unordered_map<int, int> maple;
        int ans = 0;
        for(auto answer: answers)
        {
            if(answer == 0)
                ans += 1;
            else
                maple[answer] += 1;
        }
        for(auto &i: maple)
        {
            if(i.second <= i.first + 1)
                ans += i.first + 1;
            else
            {
                int flag = i.second%(i.first + 1)!=0 ? 1 : 0;
                ans += (i.second / (i.first + 1) + flag)*(i.first + 1);
            }
        }
            
        return ans;
    }
};

File: Python/781. Rabbits in Forest.py
class Solution:
    def numRabbits(self, answers):
        """
        :type answers: List[int]
        :rtype: int
        """
        maple = dict()
        ans = 0
        for a in answers:
            if a == 0:
                ans += 1
            else:
                maple[a] = maple.get(a, 0) + 1
        for k in maple:
            if maple[k] <= k+1:
                ans += k + 1
            else:
                ans += (maple[k] // (k+1)) * (k + 1)
                if maple[k] % (k + 1) != 0:
                    ans += k + 1
        return ans